src/controllers/continuousUpdates/continuousUpdates.ts

```diff
@@ -0,0 +1,206 @@
+    }, ACCOUNT_STATE_PENDING_INTERVAL)
+
+    this.accountsOpsStatusesInterval = createRecurringTimeout(async () => {
+      const { newestOpTimestamp } = await this.#main.updateAccountsOpsStatuses()
+      // Schedule the next update only when the previous one completes
+      const interval = getIntervalRefreshTime(ACTIVITY_REFRESH_INTERVAL, newestOpTimestamp)
+      this.accountsOpsStatusesInterval.updateTimeout({ timeout: interval })
+    }, ACTIVITY_REFRESH_INTERVAL)
+
+    /**
+     * Update failed network states more often. If a network's first failed
+     *  update is just now, retry in 8s. If it's a repeated failure, retry in 20s.
+     */
+    this.fastAccountStateReFetchTimeout = createRecurringTimeout(async () => {
+      const failedChainIds = getNetworksWithFailedRPC({ providers: this.#main.providers.providers })
+      if (!failedChainIds.length) this.fastAccountStateReFetchTimeout.stop()
+
+      const retriedFastAccountStateReFetchForNetworks =
+        this.#retriedFastAccountStateReFetchForNetworks
+
+      // Delete the network ids that have been successfully re-fetched so the logic can be re-applied
+      // if the RPC goes down again
+      if (retriedFastAccountStateReFetchForNetworks.length) {
+        retriedFastAccountStateReFetchForNetworks.forEach((chainId, index) => {
+          if (!failedChainIds.includes(chainId)) {
+            delete retriedFastAccountStateReFetchForNetworks[index]
+
+            const network = this.#main.networks.networks.find(
+              (n) => n.chainId.toString() === chainId
+            )
+            this.#main.updateSelectedAccountPortfolio({ networks: network ? [network] : undefined })
+          }
+        })
+      }
+
+      // Filter out the network ids that have already been retried
+      const recentlyFailedNetworks = failedChainIds.filter(
+        (id) => !this.#retriedFastAccountStateReFetchForNetworks.find((chainId) => chainId === id)
+      )
+
+      const updateTime = recentlyFailedNetworks.length ? 8000 : 20000
+
+      await this.#main.accounts.updateAccountStates(
+        this.#main.selectedAccount.account?.addr,
+        'latest',
+        failedChainIds.map((id) => BigInt(id))
+      )
+      // Add the network ids that have been retried to the list
+      failedChainIds.forEach((id) => {
+        if (retriedFastAccountStateReFetchForNetworks.includes(id)) return
+
+        retriedFastAccountStateReFetchForNetworks.push(id)
+      })
+
+      if (!failedChainIds.length) this.fastAccountStateReFetchTimeout.stop()
+
+      this.fastAccountStateReFetchTimeout.updateTimeout({ timeout: updateTime })
+    }, 8000)
+
+    this.#main.onUpdate(() => {
+      if (this.#main.statuses.signAndBroadcastAccountOp === 'SUCCESS') {
+        this.accountStatePendingInterval.start({ timeout: ACCOUNT_STATE_PENDING_INTERVAL / 2 })
+        this.accountsOpsStatusesInterval.start()
+      }
+    }, 'continuous-update')
```

Comment:
When a transaction is broadcast, we have to stop the latest account state interval and start this one. When the pending interval is stopped, we have to start the latest one.